diff --git a/tests/env-ddf-template b/tests/env-ddf-template
index 90d7272..e43e384 100644
--- a/tests/env-ddf-template
+++ b/tests/env-ddf-template
@@ -3,7 +3,7 @@ sha1_sum() {
 }
 
 get_rootdev() {
-    local dev=$(stat -c %D /)
+    local dev=$(stat -c %D /var/tmp)
     local maj=$(expr $dev : '\(..*\)..')
     local min=${dev#$maj}
     local bd=/dev/$(basename $(readlink /sys/dev/block/$((0x$maj)):$((0x$min))))
