diff --git a/testcases/kernel/sched/hyperthreading/ht_affinity/ht_affinity.c b/testcases/kernel/sched/hyperthreading/ht_affinity/ht_affinity.c
index f6e9f27..ae530e6 100644
--- a/testcases/kernel/sched/hyperthreading/ht_affinity/ht_affinity.c
+++ b/testcases/kernel/sched/hyperthreading/ht_affinity/ht_affinity.c
@@ -20,6 +20,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sched.h>
 #include "test.h"
 
 char *TCID = "smt_smp_affinity";
@@ -48,7 +49,7 @@ len - length in bytes of the bitmask pointed to by user_mask_ptr.
 
 int HT_SetAffinity(void)
 {
-	unsigned int mask;
+	__cpu_mask mask;
 	pid_t pid;
 	int result = 1;
 	int cpu_count, i, j, k, cpuid;
@@ -147,7 +148,7 @@ unsigned long get_porc_affinity(pid_t pid)
 
 int HT_GetAffinity(void)
 {
-	unsigned int mask[2], mask1[2];
+	__cpu_mask mask[__CPU_SETSIZE / __NCPUBITS], mask1[__CPU_SETSIZE / __NCPUBITS];
 	pid_t pid;
 
 	mask[0] = 0x1;
