diff --git a/src/bench.c b/src/bench.c
index 08f3cb0..a210bfe 100644
--- a/src/bench.c
+++ b/src/bench.c
@@ -149,6 +149,10 @@ static void worker_main(void *arg)
         if (bench->ops.post_work)
                 err = bench->ops.post_work(worker);
 err_out:
+        if (!worker->id)
+                bench->start = 1;
+
+        worker->ready = 1;
         worker->ret = err;
         worker->usecs = e_us - s_us;
         wmb();
diff --git a/bin/run-fxmark.py b/bin/run-fxmark.py
index a5fe0aa..93a0783 100755
--- a/bin/run-fxmark.py
+++ b/bin/run-fxmark.py
@@ -128,6 +128,7 @@ class Runner(object):
             "btrfs":"-f",
             "jfs":"-q",
             "reiserfs":"-q",
+            "f2fs": "-f",
         }

         # media config
