diff --git a/testcases/kernel/hotplug/cpu_hotplug/include/cpuhotplug_hotplug.sh b/testcases/kernel/hotplug/cpu_hotplug/include/cpuhotplug_hotplug.sh
index 91d2a19..9f856e5 100644
--- a/testcases/kernel/hotplug/cpu_hotplug/include/cpuhotplug_hotplug.sh
+++ b/testcases/kernel/hotplug/cpu_hotplug/include/cpuhotplug_hotplug.sh
@@ -150,8 +150,8 @@ get_present_cpus()
             range_low=$(echo $part | cut -d - -f 1)
             range_high=$(echo $part | cut -d - -f 2)
         else
-            range_low=$(part)
-            range_high=$(part)
+            range_low=$part
+            range_high=$part
         fi
         for cpu in $(seq $range_low $range_high); do
             if [ -e /sys/devices/system/cpu/cpu$cpu ]; then
