diff --git a/cases/stress/hwpoison/tools/fs-metadata/fs-metadata.sh b/cases/stress/hwpoison/tools/fs-metadata/fs-metadata.sh
index 707e4a8..115d26b 100755
--- a/cases/stress/hwpoison/tools/fs-metadata/fs-metadata.sh
+++ b/cases/stress/hwpoison/tools/fs-metadata/fs-metadata.sh
@@ -116,13 +116,13 @@ function k_result()
 
 function check_disk_space()
 {
-	run_quiet cd $K_CWD 
+	run_quiet cd $K_CWD/../hwpoison
 	local depth=$(expr $1 + 1)
 	local width=$2
 	local thread=$3
 	local tree_size=$(echo "scale=2; (1-$width^$depth)/(1-$width) *2*4/1024 " | bc)
 	local total_size=$(echo "scale=2; $tree_size * 2 * $thread " | bc)
-	local free_space=$( df . -m  | awk '{ print $3}' | tail -1)
+	local free_space=$( df . -m  | awk '{ print $4}' | tail -1)
 
 	k_log "The k-tree size is $tree_size M, the total free space reqirements is about $total_size M."
 
