From ead32ba5915bfb3e5be8eef63606d1ce787b4bed Mon Sep 17 00:00:00 2001
From: Li Zhijian <lizhijian@cn.fujitsu.com>
Date: Fri, 13 Oct 2017 17:30:15 +0800
Subject: [PATCH PM-QA] thermal/thermal_06.sh: add missing get_os before check the
 return value

Signed-off-by: Li Zhijian <lizhijian@cn.fujitsu.com>
---
 thermal/thermal_06.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/thermal/thermal_06.sh b/thermal/thermal_06.sh
index d8bcd23..6fdc469 100755
--- a/thermal/thermal_06.sh
+++ b/thermal/thermal_06.sh
@@ -50,6 +50,7 @@ check_trip_point_change() {
     trip_type=0
     trip_type_path=0
     $CPU_HEAT_BIN &
+    get_os
     if [ $? -eq 1 ]; then
         cpu_pid=$(ps | grep heat_cpu| awk '{print $1}')
     else
-- 
2.15.0.rc0

