diff --git a/UnixBench/Makefile b/UnixBench/Makefile
index 880743a..e1243a4 100644
--- a/UnixBench/Makefile
+++ b/UnixBench/Makefile
@@ -86,9 +86,9 @@ OPTON = -O3 -ffast-math
 ## OS detection.  Comment out if gmake syntax not supported by other 'make'.
 OSNAME:=$(shell uname -s)
 
-ifeq ($(OSNAME),Linux)
-OPTON += -march=native -mtune=native
-endif
+# ifeq ($(OSNAME),Linux)
+# OPTON += -march=native -mtune=native
+# endif
 ifeq ($(OSNAME),Darwin)
 # (adjust flags or comment out this section for older versions of XCode or OS X)
 # (-mmacosx-versin-min= requires at least that version of SDK be installed)
diff --git a/UnixBench/Run b/UnixBench/Run
index a16a5c9..4af969b 100755
--- a/UnixBench/Run
+++ b/UnixBench/Run
@@ -809,7 +809,7 @@ sub getSystemInfo {
     }
 
     # Get graphics hardware info.
-    $info->{'graphics'} = getCmdOutput("3dinfo | cut -f1 -d\'(\'");
+    #$info->{'graphics'} = getCmdOutput("3dinfo | cut -f1 -d\'(\'");
 
     # Get system run state, load and usage info.
     $info->{'runlevel'} = getCmdOutput("who -r | awk '{print \$3}'");
@@ -1823,7 +1823,7 @@ sub main {
         $tests = $index;
     }
 
-    preChecks();
+    #preChecks();
     my $systemInfo = getSystemInfo();
 
     # If the number of copies to run was not set, set it to 1
