diff --git a/include/builddefs.in b/include/builddefs.in
index 24f838f..0a83a6d 100644
--- a/include/builddefs.in
+++ b/include/builddefs.in
@@ -35,7 +35,7 @@ PKG_DISTRIBUTION= @pkg_distribution@
 PKG_SBIN_DIR    = @sbindir@
 # A bit of a hack; by rights only state should probably go here
 # But for now ...
-PKG_LIB_DIR     = /var/lib/@pkg_name@
+PKG_LIB_DIR     = /lkp/benchmarks/@pkg_name@
 
 CC              = @cc@
 AWK             = @awk@
